test/functional/url_parser_tests.js

```diff
@@ -828,8 +953,107 @@ describe('Url Parser', function() {
       requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
     },
     test: function() {
-      var object = parse('mongodb://localhost/', { readPreference: 'secondary' });
-      expect(object.db_options.readPreference).to.equal('secondary');
+      parse('mongodb://localhost/', { readPreference: 'secondary' }, function(err, object) {
+        expect(err).to.be.null;
+        expect(object.db_options.readPreference).to.equal('secondary');
+      });
+    }
+  });
+});
+
+describe('Url SRV Parser', function() {
+  /**
+   * @ignore
+   */
+  it('should error if port is included in SRV URL', {
+    metadata: {
+      requires: { topology: ['replicaset'] }
+    },
+    test: function() {
+      parse('mongodb+srv://test6.test.build.10gen.cc:27017', {}, function(err) {
+        expect(err).to.exist;
+      });
+    }
+  });
+
+  /**
+   * @ignore
+   */
+  it('should error if no records are found in SRV discovery', {
+    metadata: {
+      requires: { topology: ['replicaset'] }
+    },
+    test: function() {
+      // This url has no srv records
+      parse('mongodb+srv://server.mongodb.com', {}, function(err) {
+        expect(err).to.exist;
+      });
+    }
+  });
+
+  /**
+   * @ignore
+   */
+  it('should allow for multiple SRV records', {
+    metadata: {
+      requires: { topology: ['replicaset'] }
+    },
+    test: function() {
+      // This url has 2 srv records, no txt records
+      // mongodb://localhost.build.10gen.cc:27018,localhost.build.10gen.cc:27017
+      parse('mongodb+srv://test1.test.build.10gen.cc', {}, function(err, object) {
+        if (err) return console.log(err);
+        expect(err).to.be.null;
+        expect(object).to.exist;
+        expect(object.servers.length).to.equal(2);
```

Comment:
chai actually can help here! you would do:
```
expect([1, 2, 3]).to.have.members([2, 1, 3]);
```

It will fuzzy match the array contents